### PR TITLE
Add palindrome charm

### LINKED ISSUE #3095: Please can 🦋 palindromic sat numbers be added as a charm?
These are by far the most sought after exotic sats (i.e. non-Rodarmor rarity sats) and are also the clear leader of the "digit rarity" class of satributes: https://x.com/ZedZeroth/status/1710287036848934984

Thank you :)

@cbspears - Do you agree? :)

### PR DESCRIPTION
**Pull request by casey**
Fixes #3095.

### CODE REVIEW

File: crates/ordinals/src/sat.rs [hunk 1]
```diff
@@ -29,6 +29,18 @@ impl Sat {
     self.n() >= 50 * COIN_VALUE * 9 && self.n() < 50 * COIN_VALUE * 10
   }
 
+  pub fn palendrome(self) -> bool {
```
Comment (raphjaph) [inline crates/ordinals/src/sat.rs:32]:
```suggestion
  pub fn palindrome(self) -> bool {
```

File: crates/ordinals/src/charm.rs [hunk 2]
```diff
@@ -90,6 +93,7 @@ impl Display for Charm {
         Self::Lost => "lost",
         Self::Mythic => "mythic",
         Self::Nineball => "nineball",
+        Self::Palendrome => "palendrome",
```
Comment (raphjaph) [inline crates/ordinals/src/charm.rs:96]:
```suggestion
        Self::Palindrome => "palendrome",
```

File: crates/ordinals/src/charm.rs [hunk 3]
```diff
@@ -15,17 +15,19 @@ pub enum Charm {
   Vindicated = 10,
   Mythic = 11,
   Burned = 12,
+  Palendrome = 13,
```
Comment (raphjaph) [inline crates/ordinals/src/charm.rs:18]:
```suggestion
  Palindrome = 13,
```

**Review by raphjaph — CHANGES_REQUESTED** [review]
This should probably bump the schema version

**Review by raphjaph — APPROVED** [review]
LGTM

**Comment by Kosi23** [comment]
How do I do this?

**Pull request merged**